src/netlink.c

```diff
@@ -512,13 +513,17 @@ static int meth_receivefrom_generic_nflua(lua_State *L) {
         return 2;
     }
 
+    struct nlattr *nl_na = (struct nlattr *) GENLMSG_DATA(nl->nlgb);
+
     if (nl->nlgb->n.nlmsg_type == NLMSG_ERROR) {
+        char buf[255];
+        memset(buf, 0, sizeof(buf));
+        payload_size = snprintf(buf, "received message error: %s", (char *)NLA_DATA(nl_na), sizeof(buf));
```

Comment:
It'd be simpler to use [`lua_pushfstring`](http://www.lua.org/manual/5.3/manual.html#lua_pushfstring)